### ISSUE TITLE
consider unused /* globals */ as unused-vars

### ISSUE THREAD

**Issue by tel** [eslint/eslint#2827]
If I create a file 

```javascript
/* globals Foo */
```

then I receive no error on not using my declared global `Foo`. This may be a little controversial since it's not entirely clear that `globals` introduces variable declarations, but without the ability to check like this the `globals` list can easily grow larger than is actually needed. Including the check would ensure that the globals list declares exactly and only the global variables needed for the file.

**Comment by gyandeeps** [comment]
They are treated that way.
What version you eslint are you using?
What kind of parser?
Can you share the actual code and the output?

**Comment by tel** [comment]
I can replicate it almost exactly as shown above:

```
$ eslint --version                                                                                                                                                        v0.22.1
$ echo "{\"rules\": { \"no-unused-vars\": 2 }}" > .eslintrc                                                                                                               $ echo "/* globals Foo */" > foo.js
$ eslint foo.js
$
```

This reports no errors, but I would expect it to complain.

**Comment by gyandeeps** [comment]
So in your case the file foo.js has nothing in it. expect the global comment. is that correct?

**Comment by tel** [comment]
Yes, that is correct.


—

**Comment by gyandeeps** [comment]
What version you eslint are you using?
What kind of parser?

**Comment by gyandeeps** [comment]
I am not able to reproduce it with online demo.
I will try it on my machine.

**Comment by tel** [comment]
Oh, apologies, I left out that line. Here's a complete trace.

```
$ mkdir eslint-globals
$ cd eslint-globals
$ echo "/* globals Foo */" > foo.js
$ eslint foo.js
$ echo "{\"rules\": {\"no-unused-vars\": 2}}" > .eslintrc
$ eslint foo.js
$ eslint --version
v0.22.1
```

**Comment by gyandeeps** [comment]
Works fine for me.

**Comment by tel** [comment]
Strange, I realize perhaps I just should update the version.

**Comment by gyandeeps** [comment]
I tried it on the version you are on too. 0.22.1

**Comment by tel** [comment]
Strange—is there any other way there might be an interaction going on?

**Comment by gyandeeps** [comment]
can you run eslint like this and provide the output
```sh
DEBUG=* eslint foo.js
```

**Comment by tel** [comment]
To confirm, I just upgraded eslint to 0.23 and can confirm that the following occurs

```
$ cat foo.js
/* global Foo */

var Foo;
$ eslint foo.js
foo.js
  3:4  error  Foo is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)

```

So eslint itself will report errors (even `no-unused-vars` errors)

**Comment by tel** [comment]
```
$ DEBUG=* eslint foo.js
  eslint:cli Running on files +0ms
  eslint:cli-engine Processing foo.js +6ms
  eslint:cli-engine Linting /Users/tel/tmp/eslint-globals/foo.js +2ms
  eslint:config Constructing config for /Users/tel/tmp/eslint-globals/foo.js +0ms
  eslint:config Using .eslintrc and package.json files +0ms
  eslint:config Loading /Users/tel/tmp/eslint-globals/.eslintrc +2ms
  eslint:config Using /Users/tel/tmp/eslint-globals/.eslintrc +4ms
  eslint:config Loading /Users/tel/tmp/.eslintrc +0ms
  eslint:config Using /Users/tel/tmp/.eslintrc +2ms
  eslint:config Merging command line environment settings +1ms
  eslint:cli-engine Load plugin react +10ms
```

**Comment by gyandeeps** [comment]
There it ids I think.
Please check what kind of configuration you have in .eslintrc file sitting in your tel/tmp directory.
its getting used also.

**Comment by tel** [comment]
Oh, that appears to be the problem indeed. Deleting `~/tmp/.eslintrc` fixes the issue. Sorry for the ignorance here! I'll try to debug that `.eslintrc` file.

**Comment by tel** [comment]
Oh, that's embarrassing. I've been picking up `.eslintrc`s containing `"no-unused-vars": [2, {"vars": "local"}]`, but I also realized *why* those existed. If I write a file like

```
/* global React:true */
React = require("react/addons");
```

then I'd like to have this pass through eslint, but without `"vars": "local"` it will not—it considers `React` to be an unused variable. Then, if I eliminate the `global` comment I get another eslint error from `no-undef`.

Any idea how to make this work?

**Comment by tel** [comment]
In other words, I want "used" to include "assigned to" for my globals, I suppose.

**Comment by tel** [comment]
Tracking this under a new issue, #2830
